### PR TITLE
Add core.compare_block_status function

### PR DESCRIPTION
**Pull request by SmallJoker**
Counter-proposal to #10897.

Intention:
 * Extensible API (for emerge or database checks later on?)
 * API support checking (`nil` return value)
 * Easy mapblock condition requirement checks in mods

The goal is not to return the direct mapblock status, but to provide a simple way to check whether a mapblock meets the required condition.

Intended use in mods
```Lua
if minetest.compare_block_status(pos, "active") == true then
	-- Supplementary action which does not depend on ABMs
else
	-- skip until block is loaded
end
```

## To do

This PR is Ready for Review.

## How to test

```Lua
core.register_chatcommand("bs", {
	func = function(name, param)
		local pos = core.string_to_pos(param)
		if not pos then
			return false, "Cannot parse pos"
		end
		return true, "Status of block " .. core.pos_to_string(pos) .. ": "
			.. tostring(core.compare_block_status(pos, "loaded"))
	end,
})

minetest.register_on_generated(function(minp, maxp, blockseed)
	print("Block inside pos " .. core.pos_to_string(minp) .. ": "
		.. tostring(core.compare_block_status(minp, "emerging")))
end)
```

Run away from the original position, check different map positions.

### CODE REVIEW

**Comment by rubenwardy** [comment]
```lua
if minetest.get_block_status(pos) >= minetest.block_states.ACTIVE then
```

would be a better API imo

**Comment by SmallJoker** [comment]
> ```lua
> if minetest.get_block_state(pos) >= minetest.block_states.ACTIVE then
> ```
> 
> would be a better API imo

I wanted to keep the API easily function extensible to implement more specific states at a later time (`"stored" for DB, or "pending" when in emerge queue). That approach would not ensure that it works reliably in mods which would check against each possible state, i.e. `if state == minetest.block_states.ACTIVE or minetest.block_states.LOADED then` would break if there would ever be another state in between.

**Comment by sorcerykid** [comment]
I really like this PR, as it seems much more flexible. I definitely hope that it will be possible to check if a mapblock exists in the database or, for even more granularity, determining if it's in the queue to be emerged. Those would be very useful status checks.

I admit I'm a little confused by the comparison operators. Wouldn't it suffice for most use-cases just to return a boolean?

**Comment by sfan5** [comment]
This function could indeed just return a boolean. Callers can differ between `true`, `false` and `nil` (=unsupported).
Or can anyone think of an usecase for differing between "exactly in the wanted condition" and "surpasses the wanted condition"?

**Comment by SmallJoker** [comment]
PR updated accordingly. There's yet no "contained in database" check because I am currently not keen on doing performance benchmarks for that. However, the PR does surely cover more than the feature from #10897.

I honestly do not know what's going on with the failed tests. `--un-unittests` and the API works locally as expected.

File: doc/lua_api.txt [hunk 1]
```diff
@@ -5849,6 +5849,19 @@ Misc.
     * If `transient` is `false` or absent, frees a persistent forceload.
       If `true`, frees a transient forceload.
 
+* `minetest.compare_block_status(pos, condition)`
+    * Checks whether the mapblock at positition `pos` is in the wanted condition.
+    * `condition` may be one of the following values:
+        * `"unknown"`: not in memory
+        * `"emerging"`: in the queue for generating
```
Comment (sfan5) [inline doc/lua_api.txt:5856]:
```suggestion
        * `"emerging"`: in the queue for loading from disk or generating
```

**Review by sfan5 — APPROVED** [review]
Looks good other than that.
~~I've restarted the CI runs, they should pass now.~~ You can ignore the CI failures.

**Pull request merged**